subseas_flex_fcst/maproom.py

```diff
@@ -131,54 +131,75 @@ def write_mapTitle(startDate, leadTime, leadTimeOptions):
     return f'{targetPeriod} {CONFIG["variable"]} Forecast issued {startDate}'
 
 @APP.callback(
-    Output("cdf_graph", "figure"),
-    Output("pdf_graph", "figure"),
-    Output("map", "click_lat_lng"),
     Output("layers_group", "children"),
     Output("latInput", "value"),
     Output("lngInput", "value"),
+    Output("map", "click_lat_lng"),
     Input("submitLatLng","n_clicks"),
     Input("map", "click_lat_lng"),
-    Input("startDate","value"),
-    Input("leadTime","value"),
     State("latInput", "value"),
     State("lngInput", "value")
 )
-def local_plots(n_clicks, click_lat_lng, startDate, leadTime, latitude, longitude):
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
     # Reading
-
-    fcst_mu, fcst_var, obs, hcst = read_cptdataset(leadTime, startDate, y_transform=CONFIG["y_transform"])
+    filesNameList = glob.glob(f'{DATA_PATH}/{CONFIG["forecast_mu_filePattern"]}')
+    startDate = re.search("\w{3}-\w{1,2}-\w{4}",filesNameList[0])
+    startDate = datetime.strptime(startDate.group(),"%b-%d-%Y").strftime("%b-%-d-%Y")
+    fcst_mu, fcst_var, obs, hcst = read_cptdataset(
+        list(CONFIG["leads"])[0],
+        startDate,
+        y_transform=CONFIG["y_transform"]
+    )
     if click_lat_lng is None: #Map was not clicked
         if n_clicks == 0: #Button was not clicked (that's landing page)
-            lat = (fcst_mu.Y[0].values+fcst_mu.Y[-1].values)/2
-            lng = (fcst_mu.X[0].values+fcst_mu.X[-1].values)/2
+            lat = fcst_mu.Y[int(fcst_mu.Y.size/2)].values
+            lng = fcst_mu.X[int(fcst_mu.X.size/2)].values
         else: #Button was clicked
             lat = latitude
             lng = longitude
     else: #Map was clicked
         lat = click_lat_lng[0]
         lng = click_lat_lng[1]
-    # Errors handling
+    half_res = (fcst_mu["X"][1] - fcst_mu["X"][0]) / 2
+    tol = np.sqrt(2 * np.square(half_res)).values
     try:
-        half_res = (fcst_mu["X"][1] - fcst_mu["X"][0]) / 2
-        tol = np.sqrt(2 * np.square(half_res)).values
         nearest_grid = fcst_mu.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
         lat = nearest_grid.Y.values
         lng = nearest_grid.X.values
-        fcst_mu = fcst_mu.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
-        fcst_var = fcst_var.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
-        obs = obs.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
+    except KeyError:
+        lat = lat
+        lng = lng
+    return dlf.Marker(position=[lat, lng]), lat, lng, None
+
+
+@APP.callback(
+    Output("cdf_graph", "figure"),
+    Output("pdf_graph", "figure"),
+    Input("layers_group", "children"),
```

Comment:
Can we put the `Marker` in the layout, and have `pick_location` just set the Marker's position instead of generating the Marker itself? Then this input, which seems strange and brittle to me (couldn't `layers_group` have other children in the future?), could be replaced with that position.

Reply:
yes, agreed. I was concerned that layers_group was sort of blocked for this one purpose.

but when I try to have a dlf.Marker in the layout and pass it "position" I get:

TypeError: Required argument `position` was not specified.

when I start to think I understood something... Maybe I'll try again freshly tomorrow morning

Reply:
OK, we can come back to this after (hopefully) fixing the problem with the deferred call to the layout function.

Reply:
Looking at this with fresh eyes, this still makes me uncomfortable. Pending a change to the Marker class that allows it to be created with no position, how about we use the center of the dataset as the initial position?

Reply:
I've just pushed 2 commits, the first one is trying to do that. Not sure I got what you had in mind so please have a look.